master/internal/workspace/postgres_workspace_intg_test.go

```diff
@@ -0,0 +1,136 @@
+//go:build integration
+// +build integration
+
+package workspace
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/proto/pkg/workspacev1"
+)
+
+const (
+	cluster1 = "C1"
+	cluster2 = "C2"
+)
+
+func TestMain(m *testing.M) {
+	pgDB, err := db.ResolveTestPostgres()
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	err = db.MigrateTestPostgres(pgDB, "file://../../static/migrations", "up")
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	err = etc.SetRootPath("../../static/srv")
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	os.Exit(m.Run())
+}
+
+type Bindings struct {
+	bun.BaseModel `bun:"table:workspace_namespace_bindings"`
+	WorkspaceID   int    `bun:"workspace_id"`
+	ClusterName   string `bun:"cluster_name"`
+	NamespaceName string `bun:"namespace_name"`
+}
+
+func TestGetNamespaceFromWorkspace(t *testing.T) {
+	ctx := context.Background()
+	wksp1 := "wksp_1"
+	wkspID1, _ := db.RequireMockWorkspaceID(t, db.SingleDB(), wksp1)
+	wksp2 := "wksp_2"
+	wkspID2, _ := db.RequireMockWorkspaceID(t, db.SingleDB(), wksp2)
+	wksp3 := "wksp_3"
+	wkspID3, _ := db.RequireMockWorkspaceID(t, db.SingleDB(), wksp3)
+	cluster1 := "C1"
+	cluster2 := "C2"
+
+	b1 := Bindings{WorkspaceID: wkspID1, ClusterName: cluster1, NamespaceName: "n1"}
+	b2 := Bindings{WorkspaceID: wkspID1, ClusterName: cluster2, NamespaceName: "n2"}
+	b3 := Bindings{WorkspaceID: wkspID2, ClusterName: cluster1, NamespaceName: "n3"}
+
+	bindings := []Bindings{b1, b2, b3}
+
+	_, err := db.Bun().NewInsert().Model(&bindings).Exec(ctx)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name          string
+		binding       Bindings
+		workspaceName string
+		expectNoError bool
```

Comment:
readability nit: go tests (including ours) frequently use `wantErr`. can we match?

Reply:
done